gammapy/modeling/parameter.py

```diff
@@ -588,6 +588,34 @@ def freeze_all(self):
         for par in self._parameters:
             par.frozen = True
 
+    def get_subcovar(self, names):
```

Comment:
This is used in spatial_model.position_error as
`sub_covar = pars.get_subcovar(["lon_0", "lat_0"])
`
In such case you have to look for the parameter by name anyway and I think it is more convenient to have this method rather than filtering the objects to define a new parameters instance and then applying set_subcovar_from_parameters.

Reply:
As this method is part of the public API I would strongly recommend to only support the safe case (i.e. a `Parameters` object). Note that this is mainly a matter of documentation, because the case you've described above still (silently) works. I would just not expose it in the docstring to prevent incorrect results.

Reply:
ok removed